code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -55,6 +55,7 @@
 
 	//Hot simple_animal baby making vars
 	var/childtype = null
+	var/rarechildtype = null	//Special children
```

Comment:
Look at /obj/effect/lootspawner for an example of spawning from a weighted list. On my phone so  it'd be a pain to find/make an example for you